### ISSUE TITLE
Error message presented to user who does not have access to setup is misleading.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#9477]
Rancher server version  - v1.6.6-rc3

Have AD auth enabled with "Site Access" set to "Allow members of Environments.plus Authorized Users and Organization"

Scenario 1:
Try to log in using valid credentials as a user who does not have access to the setup.

User is presented with "Username or Password incorrect" 

```
Request URL:http://<ip address>:8080/v2-beta/token
Request Method:POST
Status Code:401 Unauthorized
Remote Address:35.193.214.172:8080
Referrer Policy:no-referrer-when-downgrade
```

This error message is misleading. Can this be error message be improved?

Scenario 2:

When invalid credentials  are passed ("Valid username but invalid password ) - User is presented with "Error Response from Auth service"

```
Request URL:http://<ip>:8080/v2-beta/token
Request Method:POST
Status Code:503 Service Unavailable
Remote Address:35.193.214.172:8080
Referrer Policy:no-referrer-when-downgrade
```

rancher-server logs:
```
time="2017-07-26T01:01:30Z" level=error msg="GetToken failed with error: Error LDAP Result Code 49 \"Invalid Credentials\": 80090308: LdapErr: DSID-0C0903CF, comment: AcceptSecurityContext error, data 52e, v2580\x00 in ldap bind" 
2017-07-26 01:01:30,020 ERROR [:] [] [] [] [tp1020391880-98] [.a.a.i.e.ExternalServiceAuthProvider] Got error from Auth service. statusCode: 500 
```

Is this error message acceptable?

**Comment by vincent99** [comment]
- 4xx codes: the client's input is unacceptable and if they try different input the request may succeed.
  - 401 Unauthorized: the credentials provided are wrong.
  - 403 Forbidden: the credentials provided are valid, but we don't want to let you in anyway.
  - 422 Unprocessable Entity: validation of the body contents failed (not really relevant to this issue)
- 5xx codes: something went wrong on the server's side and the client didn't do anything wrong and can't do anything to fix it.

So both the above sound wrong.  There are three situations:

User or password incorrect:
  - Status Code: 401 Unauthorized
  - Body: Something like `{"type": "error", "statusCode": 401, "message": "Username or password incorrect"}`
  - The UI shows it's own translation string for this based on seeing 401.

Username and password are correct, but that person is not allowed in:
  - Status Code: 403 Forbidden
  - Body: Something like `{"type": "error", "statusCode": 403, "message": "Your account does not have access"}`
  - I don't think the UI does currently, but it should show a different translation string based on seeing 403.

Unable to determine if credentials are correct because you couldn't talk to auth provider or the auth service:
  - Status Code: 500 Internal Server Error
  - Body: Something like {"type": "error", "statusCode": 500, "message": "Error reaching ActiveDirectory server"}, or "Error reaching auth service", "SSL verification failed while talking to ActiveDirectory server", or whatever you can figure out about what happened.
  - The UI should show whatever the message is (untranslated), as any number of things can be wrong.

**Comment by prachidamle** [comment]
@deniseschannon @sangeethah The above fixes address Scenario 2. Please can you log a separate issue for Scenario 1

**Comment by sangeethah** [comment]
@prachidamle , Fix for this is not going to be made for this ?
```403 Forbidden: the credentials provided are valid, but we don't want to let you in anyway.```

**Comment by prachidamle** [comment]
@sangeethah now Scenario1 is also fixed. 
Username and password are correct, but that person is not allowed in:
Status Code: 403 Forbidden
Body: Something like {"type": "error", "statusCode": 403, "message": "Your account does not have access"}

**Comment by deniseschannon** [comment]
Available in v1.6.8-rc1

**Comment by sangeethah** [comment]
Tested with Rancher server version - v1.6.8-rc1

Scenario 1:
Site Access setup as - "Allow member of Environment , plus Authorized Users and Organizations"
Try to log in using valid credentials as a user who does not have access to the setup.
User is presented with "Your account does not have access"

Scenario 2:
Site Access setup as - "Allow member of Environment , plus Authorized Users and Organizations"
Try to log in using invalid credentials (valid user name/invalid password , invalid user name)
User is presented with "Username or Password incorrect."

Scenario 3:
Site Access setup as - "Restrict access to only  Authorized Users and Organizations"
Try to log in using valid credentials as a user who does not have access to the setup.
User is presented with "Username or Password incorrect." in this case

```
time="2017-08-09T18:55:08Z" level=error msg="Cannot locate user information for (&(sAMAccountName=testuser4)(|(distinguishedName=CN=Test User1,CN=Users,DC=tad,DC=rancher,DC=io)(distinguishedName=CN=Test User2,CN=Users,DC=tad,DC=rancher,DC=io)(memberof=CN=testgroup4,CN=Users,DC=tad,DC=rancher,DC=io)(memberof=CN=testgroup3,CN=Users,DC=tad,DC=rancher,DC=io)))" 
```
Reopening this issue to fix scenario 3.

**Comment by deniseschannon** [comment]
Available with `rancher/server:v1.6.8-rc4`

**Comment by sangeethah** [comment]
Tested with rancher/server:v1.6.8-rc4:

Site Access setup as - "Restrict access to only Authorized Users and Organizations"
Try to log in using valid credentials as a user who does not have access to the setup.
User is presented with  - "Your account does not have access" as expected.
